Module '"react-native"' has no exported member 'NativeComponent'.

# Bug report

<!--
👋 Hi!

🚨 Please read the following carefully before opening a new issue. Your issue may
be closed if it doesn't provide the required pieces of information. 🚨

Before submitting a new issue, please:

- Test using the latest release of the library, as maybe your bug has been already fixed.
- Check for possible duplicate issues, with possible answers.
- Check on https://reactnative.directory if this package supports your target platform (Android, iOS, Expo, …)

Still ready? Fill the template. 👇
-->

## Summary

There is a Typescript build issue when importing DateTimePicker:

```
node_modules/@react-native-community/datetimepicker/src/index.d.ts:2:9 - error TS2724: Module '"react-native"' has no exported member 'NativeComponent'. Did you mean 'TextComponent'?

2 import {NativeComponent, ViewProps} from 'react-native';
          ~~~~~~~~~~~~~~~

  node_modules/@types/react-native/index.d.ts:932:15
    932 declare class TextComponent extends React.Component<TextProps> {}
                      ~~~~~~~~~~~~~
    'TextComponent' is declared here.


Found 1 error.
```

This appears to be due to some cleanup work done in `@types/react-native` to support 0.62
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43179

<!--
Provide a clear and concise description of what the bug is.
-->

## Environment info

<!--
Run `react-native info` in your terminal and paste the results here. Also, include the *precise* version number of this library that you are using in the project.
-->

`react-native info` output:

```bash
System:
    OS: macOS 10.15.4
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 86.73 MB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.16.3 - ~/.nvm/versions/node/v12.16.3/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.4 - ~/.nvm/versions/node/v12.16.3/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.9.1 - /Users/alockha/.rvm/gems/ruby-2.6.3/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 13.5, DriverKit 19.0, macOS 10.15, tvOS 13.4, watchOS 6.2
    Android SDK: Not Found
  IDEs:
    Android Studio: Not Found
    Xcode: 11.5/11E608c - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.6 - /usr/bin/javac
    Python: 2.7.16 - /usr/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.11.0 => 16.11.0 
    react-native: 0.62.2 => 0.62.2 
  npmGlobalPackages:
    *react-native*: Not Found
```

Library version: 2.4.2

## Steps to reproduce

<!--
- You must provide a clear list of steps and code to reproduce the problem.
- Keep the code reproducing the bug as simple as possible, with the minimum amount of code required to reproduce the issue. See https://stackoverflow.com/help/mcve.
- If this library has additional install steps, describe them (e.g., pod install? jetify? etc).
- Either re-create the bug using the repository's example app or link to a GitHub repository with code that reproduces the bug.
- Explain the steps we need to take to reproduce the issue:
-->

1. `npx react-native init DatePickerError --template react-native-template-typescript`
2. `cd DatePickerError`
3. `npm i @react-native-community/datetimepicker`
4. Import the date picker in `App.tsx`
5. Run `tsc`

Describe what you expected to happen:

1. There should not be TS errors

## Reproducible sample code

https://github.com/salockhart/RNDatePickerError

--- a/src/index.d.ts
+++ b/src/index.d.ts
@@ -1,5 +1,5 @@
 import {FC, Ref, SyntheticEvent} from 'react';
-import {NativeComponent, ViewProps} from 'react-native';
+import {NativeMethods, ViewProps} from 'react-native';
 
 type IOSMode = 'date' | 'time' | 'datetime' | 'countdown';
 type AndroidMode = 'date' | 'time';
@@ -130,7 +130,7 @@ export type DateTimePickerResult = Readonly<{
   minute: number;
 }>;
 
-export type RCTDateTimePickerNative = typeof NativeComponent;
+export type RCTDateTimePickerNative = NativeMethods;
 export type NativeRef = {
   current: Ref<RCTDateTimePickerNative> | null;
 };

